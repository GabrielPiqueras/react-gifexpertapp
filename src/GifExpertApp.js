import { useState } from "react";

import AddCategory from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

/**
 * COMPONENTE PRINCIPAL
 * 
 * 1. Guarda una lista de categorías en un array, como estado del componente.
 * 2. Muestra el componente 'AddCategory', es un formulario para añadir una nueva categoría.
 * 3. Muestra las categorías en una lista ordenada, mapeando el array y añadiendo una clave a cada una.
 * 4. Cada categoría de la lista se muestra a través del componente GifGrid, 
 * 4. Para añadir una nueva, se envía al componente hijo la función setCategories (Obligatoria), que allí permite añadir una nueva.
*/
const GifExpertApp = () => {
    
    const [categories, setCategories] = useState(['One Piece']);

    return(
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            {   
                categories.map( category  =>
                    <GifGrid key={ category } category={ category }/>
                )
            }
        </>
    );
}

export default GifExpertApp;