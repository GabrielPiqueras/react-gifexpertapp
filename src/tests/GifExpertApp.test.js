
import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el componente <GifExpertApp />', () => {
    
    test('<GifExpertApp /> se renderiza correctamente.', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {
        const categories = ['One Piece', 'Juego de Tronos', 'Arcane'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        const grids = wrapper.find('GifGrid');

        expect(wrapper).toMatchSnapshot();
        expect(grids.length).toBe(categories.length);
    }); 
});