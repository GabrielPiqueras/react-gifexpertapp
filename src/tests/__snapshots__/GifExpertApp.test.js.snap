// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> <GifExpertApp /> se renderiza correctamente. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Piece"
    key="One Piece"
  />
  <GifGrid
    category="Juego de Tronos"
    key="Juego de Tronos"
  />
  <GifGrid
    category="Arcane"
    key="Arcane"
  />
</Fragment>
`;
